### PR TITLE
Replace local tags warning with debug log and documentation

### PR DESCRIPTION
**Pull request by dgageot**
+ Improve documentation for taggers
+ Remove scary warning about local images. Move it to the debug logs and provide a pointer to the documentation instead.

### CODE REVIEW

File: docs/content/en/docs/pipeline-stages/taggers.md [hunk 1]
```diff
@@ -6,21 +6,43 @@ featureId: tagpolicy
...
+Skaffold supports multiple taggers or tag policies for tagging images:
 
-* `gitCommit`: use git commit IDs as tags
-* `sha256`: use sha256 hashes of contents as tags
-* `envTemplate`: use values of environment variables as tags
-* `dateTime`: use date and time values as tags
+ + the `gitCommit` tagger uses git commits/references to tag images.
+ + the `sha256` tagger uses `latest` to tag images.
+ + the `envTemplate` tagger uses environment variables to tag images.
+ + the `datetime` tagger uses current date and time, with a configurable pattern.
```
Comment (briandealwis) [inline docs/content/en/docs/pipeline-stages/taggers.md:14]: We should say which is the default up here.

File: docs/content/en/docs/pipeline-stages/taggers.md [hunk 2]
```diff
@@ -6,21 +6,43 @@ featureId: tagpolicy
...
-The tag policy is specified in the `tagPolicy` field of the `build` section of the
-Skaffold configuration file, `skaffold.yaml`.
+### Configuration
+
+The tag policy is specified in the `tagPolicy` field of the `build` section
+of the `skaffold.yaml` configuration file.
+
+{{% readfile file="samples/taggers/git.yaml" %}}
+
+The default tagger, if none is specified in the `skaffold.yaml`, is the `git` tagger.
```
Comment (briandealwis) [inline docs/content/en/docs/pipeline-stages/taggers.md:23]: I think this should be moved above.  And it's the `gitCommit` tagger.

File: docs/content/en/docs/pipeline-stages/taggers.md [hunk 3]
```diff
@@ -6,21 +6,43 @@ featureId: tagpolicy
...
 For a detailed discussion on Skaffold configuration, see
 [Skaffold Concepts]({{< relref "/docs/design/config.md" >}}) and
 [skaffold.yaml References]({{< relref "/docs/references/yaml" >}}).
 
-## `gitCommit`: uses Git commit IDs as tags
+### How tagging works
+
+ + Image tags are computed before the images are built.
+ + No matter the tagger, Skaffold always uses immutable references in Kubernetes manifests.
+   Which reference is used depends on whether the images are pushed or not:
```
Comment (briandealwis) [inline docs/content/en/docs/pipeline-stages/taggers.md:33]:
```suggestion
   Which reference is used depends on whether the images are pushed to a registry or loaded directly into the cluster (such as via the Docker daemon):
```

File: docs/content/en/docs/pipeline-stages/taggers.md [hunk 4]
```diff
@@ -6,21 +6,43 @@ featureId: tagpolicy
...
+     + **When images are pushed**, their immutable digest is available. Skaffold then references
+       images both by tag and digest. Something like `image:tag@sha256:abacabac...`.
+       Using both the tag and the digest seems superfluous but it guarantees immutability
+       and helps users quickly see which version of the image is used.
+     + **When images are not pushed**, digests are not available. We have the tags and the
```
Comment (briandealwis) [inline docs/content/en/docs/pipeline-stages/taggers.md:38]:
```suggestion
     + **When images are loaded directly into the cluster**, such as loading into the cluster's Docker daemon, digests are not available. We have the tags and the
```

File: pkg/skaffold/runner/deploy.go [hunk 5]
```diff
@@ -41,7 +43,9 @@ func (r *SkaffoldRunner) Deploy(ctx context.Context, out io.Writer, artifacts []
 	}
 
 	if r.imagesAreLocal && len(artifacts) > 0 {
-		color.Green.Fprintln(out, "   local images can't be referenced by digest. They are tagged and referenced by a unique ID instead")
+		logrus.Debugln("Local images can't be referenced by digest.")
```
Comment (briandealwis) [inline pkg/skaffold/runner/deploy.go:46]: This should be a single message.  But is it actually necessary?  It's not triggered by anything that the user has done.

Comment (dgageot) [reply pkg/skaffold/runner/deploy.go:46]: This is helpful to understand why the tags are not used locally.

**Pull request merged**